consensus/src/processes/parents_builder.rs

```diff
@@ -183,5 +183,311 @@ impl<T: HeaderStoreReader, U: ReachabilityStoreReader, V: RelationsStoreReader>
 
 #[cfg(test)]
 mod tests {
-    // TODO: add unit-tests for calc_block_parents
+    use std::{
+        collections::{HashMap, HashSet},
+        sync::Arc,
+    };
+
+    use crate::{
+        model::{
+            services::reachability::MTReachabilityService,
+            stores::{
+                errors::StoreError,
+                headers::{HeaderStoreReader, HeaderWithBlockLevel},
+                reachability::MemoryReachabilityStore,
+                relations::RelationsStoreReader,
+            },
+        },
+        processes::reachability::tests::{DagBlock, DagBuilder},
+    };
+
+    use super::ParentsManager;
+    use consensus_core::{
+        blockhash::{BlockHashes, ORIGIN},
+        header::Header,
+    };
+    use hashes::Hash;
+    use itertools::Itertools;
+    use parking_lot::RwLock;
+
+    struct HeaderStoreMock {
```

Comment:
Seems like a classic place for a mock lib. Should we add one? What do you think @elichai ? The downside of https://docs.rs/mockall/latest/mockall/ is that you need to add `#[automock]` annotation to the original trait.